### PR TITLE
secrets/mysql: Add `tls_server_name` and `tls_skip_verify` parameters

### PR DESCRIPTION
**Pull request by jasonodonnell**
The MySQL implementation supports TLS verification and auth, however, when testing against Cloud SQL it was very painful to verify the CA. This is because Cloud SQL does not add IP SANs to the certificate. The SANs available reference internal VM names which aren't easily obtainable as a user. I've added these parameters to make the engine more flexible. Although the connection url does support these, the MySQL driver stores a tls.Config and references it by name in the connection URL, so it makes sense to add these there.

### CODE REVIEW

File: plugins/database/mysql/connection_producer.go [hunk 1]
```diff
@@ -111,12 +112,12 @@ func (c *mySQLConnectionProducer) Init(ctx context.Context, conf map[string]inte
 	c.Initialized = true
 
 	if verifyConnection {
-		if _, err := c.Connection(ctx); err != nil {
-			return nil, fmt.Errorf("error verifying connection: %w", err)
+		if _, err = c.Connection(ctx); err != nil {
+			return nil, fmt.Errorf("error verifying connection connection: %w", err)
```
Comment (swenson) [inline plugins/database/mysql/connection_producer.go:116]: This error message might be slightly confusing confusing.
```suggestion
			return nil, fmt.Errorf("error verifying - connection: %w", err)
```

File: plugins/database/mysql/connection_producer.go [hunk 2]
```diff
@@ -111,12 +112,12 @@ func (c *mySQLConnectionProducer) Init(ctx context.Context, conf map[string]inte
...
 		}
 
 		if err := c.db.PingContext(ctx); err != nil {
-			return nil, fmt.Errorf("error verifying connection: %w", err)
+			return nil, fmt.Errorf("error verifying connection ping: %w", err)
```
Comment (swenson) [inline plugins/database/mysql/connection_producer.go:120]:
```suggestion
			return nil, fmt.Errorf("error verifying - ping: %w", err)
```

**Review by swenson — APPROVED** [review]
LGTM; a suggestion.

Were you able to test this?

File: plugins/database/mysql/connection_producer.go (hunk 2 above)
Comment (jasonodonnell) [reply plugins/database/mysql/connection_producer.go:120]: Oops, these were debug lines but I do like the improvement. Thanks.

**Comment by jasonodonnell** [comment]
> LGTM; a suggestion.
> 
> Were you able to test this?

@swenson Yup, tested it against an mTLS Google Cloud SQL instance. We have existing tests, however, they don't run in Circle and adding mTLS support to the test wasn't straight forward.

**Review by austingebauer — APPROVED** [review]

**Pull request merged**